src/iris/ui/static/js/iris.js

```diff
@@ -2213,6 +2356,45 @@ iris = {
       }
       this.createPriorityTable();
     },
+    addApplicationNotification: function() {
+      var app = this.data.$addAppNotificationSelect.val();
+
+      this.data.$addAppNotificationSelect.val('');
+
+      if(this.data.settings.customApp.indexOf(app) === -1){
+        this.data.settings.customApp.push(app);
+      }
+      this.createCustomNotificationTable();
```

Comment:
The table entry for the app (on the main user settings page) appears before you click "Save" in the modal, and persists even if you cancel out of it. This entry should only appear after clicking "Save".